src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -44,24 +59,70 @@ def __init__(self, msConfig, logger=None):
         self.msConfig.setdefault("enableDataPlacement", False)
         self.msConfig.setdefault("excludeDataTier", ['NANOAOD'])
         self.msConfig.setdefault("rucioAccount", 'wma_test')
+        self.msConfig.setdefault("mongoDBUrl", 'mongodb://localhost')
+        self.msConfig.setdefault("mongoDBPort", 8230)
+        self.msConfig.setdefault("streamerBufferFile", "/tmp/msOutput/requestRecords")
         self.uConfig = {}
         self.emailAlert = EmailAlert(self.msConfig)
 
+        self.cric = CRIC(logger=self.logger)
+        self.uConfig = {}
+        self.campaigns = {}
+        self.psn2pnnMap = {}
+
+        msOutIndex = IndexModel('RequestName', unique=True)
+        msOutDBConfig = {
+            'database': 'msOutDB',
+            'server': self.msConfig['mongoDBUrl'],
+            'port': self.msConfig['mongoDBPort'],
+            'logger': self.logger,
+            'create': True,
+            'collections': [
+                ('msOutRelValColl', msOutIndex),
+                ('msOutNonRelValColl', msOutIndex)]}
+
+        self.msOutDB = MongoDB(**msOutDBConfig).msOutDB
```

Comment:
Oook, this one is tricky. And I agree it is not perfect, I do not like it either. So things start from how Mongo is creating a database session: [1]. Nothing wrong with that of course, but as you may have already noticed, the `MongoDB` class does not inherit directly from `pymongo.MongoClient` (which I should have done from the very beginning ... but any way). So now the actual database object is attached to the `MongoDB` instance as an attribute here [2]. (and because of the aforementioned lack of inheritance, even the client call is not just `self['SomedbName']` as one should expect, but it is `self.cleint['SomedbName']` instead - because the client is also attached to the object as an attribute). So this is how I ended up with that call:
```
dbObject = MongoDB(database='dbObject').dbObject
```
*NOTE:* Please notice - it will break if the database string in this call is not the same as the variable name - I know this is horrible.

I could have updated the object's __dict__ also so we can have calls similar to the the one that are implemented in the MongoClient. But it would just have lead to repetitions in the call. It would look something like:

```
 dbObject = MongoDB(database='dbObject')['dbObject']
```
As you see - not much better. And internaly in the class the thing would look even worse. 

My Opinion is that this 'MongoDB' module, even as good as it is right now, will still need a refactoring in the next cycle. So an issue combining this one here and the a fore mentioned 'moving the Database instance on top of the thread' is a must for the next month I would say.

[1]
https://api.mongodb.com/python/3.4.0/tutorial.html#getting-a-database 

[2]
https://github.com/dmwm/WMCore/blob/eb2edafab1639a248d1771a1a344e0317710355e/src/python/WMCore/Database/MongoDB.py#L123

Reply:
I would rather go for making our Class inheriting from Mongo Client and test if everything works properly which would make us quite aligned with the behavior of the upstream provided tools and just enriching the object with the methods we need for our specific task.
And yes, we'd better follow this on a separate issue - I am refering it here again: https://github.com/dmwm/WMCore/issues/9711

Reply:
Todor, I'm honestly lost in all these changes, but please do not make WMCore classes inherit from 3rd party APIs as you have no control over 3rd party libraries/APIs, and if they change their class behavior your class will misbehave as well without any notice.

I also don't see any point wrapping out Mongo APIs with WMCore APIs for the same reason. And, I rather will use 3rd party APIs directly which can be easy to fix in a long run.

Reply:
Valentin has a good point about inheriting from a 3rd party library.
About the wrapper, I'm all in favor of it because then we can put all the extra necessary logic/exception handling/etc around the client APIs and keep a consistent usage of them.

Reply:
Thanks @vkuznet and @amaltaro  I can see your point and Agree with that. I wont make it inherit from the Mongo Client, but will try few other things in order to the object construction.